openerp-client/src/main/java/org/bahmni/openerp/web/request/builder/RequestBuilder.java

```diff
@@ -22,6 +23,7 @@ public static String buildNewXMLRequest(OpenERPRequest openERPRequest, Object id
             velocityEngine.init();
             Template template = velocityEngine.getTemplate("request/template/xml_template.vm");
             VelocityContext context = new VelocityContext();
+            context.put("esc", new EscapeTool());
```

Comment:
What does this line of code do?

Reply:
The Escape tool is an apache tool that helps escape characters in the Velocity Engine. Previously, the Parameter class did a StringEscapeUtils.escapeXML() on all our values, which was causing issues with our new json payload. So we removed the type casting and decided to handle it in the velocity template instead. The `esc` that we have set enables the value to be set to XML supported escape characters when we build the request from the template. 
This gave us more flexibility as compared to how it was done before.

Reply:
Please add this description / hyperlink as a comment in code, since its not obvious why you added this.